File: hgctl/src/commands/context/use.ts
import { Command } from '@commander-js/extra-typings';
import { setCurrentContext } from '../../config/context';
import { Logger } from '../../logger';

export function setupUseCommand(parent: Command) {
  parent
    .command('use <context-name>')
    .description('Switch to a different context')
    .action(async (contextName, _options, command) => {
      const globalOpts = command.optsWithGlobals();
      const logger = globalOpts.logger as Logger;
      
      try {
        await setCurrentContext(contextName);
        logger.info(`Switched to context "${contextName}"`);
      } catch (error) {
        logger.error(`Failed to switch context: ${(error as Error).message}`);
        process.exit(1);
      }
    });
}

File: hgctl-go/internal/commands/context/use.go
package context

import (
	"fmt"

	"github.com/urfave/cli/v2"
	"go.uber.org/zap"

	"github.com/Layr-Labs/hourglass-monorepo/hgctl-go/internal/config"
)

func useCommand() *cli.Command {
	return &cli.Command{
		Name:      "use",
		Usage:     "Switch to a different context",
		ArgsUsage: "<context-name>",
		Action:    contextUseAction,
	}
}

func contextUseAction(c *cli.Context) error {
	if c.NArg() != 1 {
		return cli.ShowSubcommandHelp(c)
	}

	contextName := c.Args().Get(0)
	log := config.LoggerFromContext(c.Context)

	cfg, err := config.LoadConfig()
	if err != nil {
		return fmt.Errorf("failed to load config: %w", err)
	}

	if _, exists := cfg.Contexts[contextName]; !exists {
		return fmt.Errorf("context '%s' not found", contextName)
	}

	cfg.CurrentContext = contextName

	if err := config.SaveConfig(cfg); err != nil {
		return fmt.Errorf("failed to save config: %w", err)
	}

	log.Info("Switched context", zap.String("context", contextName))
	fmt.Printf("Switched to context '%s'\n", contextName)

	return nil
}
